### PR TITLE
about:preferences now has "autohide menu" option

### PR DESCRIPTION
**Pull request by bsclifton**
Updates to resolve https://github.com/brave/browser-laptop/issues/1524

- Home button and this new setting pulled into an "Appearance settings" area of the general tab.
- Change takes effect immediately

~~What is missing (will address in future PR):~~
- ~~context menu item in the bookmarks toolbar~~
- ~~default this new option to "false", but only for Windows users~~

### CODE REVIEW

File: app/extensions/brave/locales/en-US/menu.properties [hunk 1]
```diff
@@ -120,3 +120,4 @@ zoom=Zoom
 new=New
 learnSpelling=Learn Spelling
 ignoreSpelling=Ignore Spelling
+autoHideMenuBar=Hide the menu bar
```
Comment (bbondy) [inline app/extensions/brave/locales/en-US/menu.properties]: nit: Hide Menu Bar

**Comment by bbondy** [comment]
This is looking good but the main thing I see before this can be merged in is that there should be no option for this at all for OS X (process.platform === 'darwin')

**Comment by bsclifton** [comment]
Good catch, thanks :smiley: I'll tweak this PR to hide the preference (and the context menu) on OS X

**Comment by bsclifton** [comment]
Screenshots (NOTE: changes only show for Windows / Linux)

New option in preferences


New context menu item when right clicking on bookmarks toolbar

**Comment by bbondy** [comment]
Manually merged because of a conflict, thanks!

**Comment by luixxiul** [comment]
On Ubuntu Hide Menu Bar seems to hide nothing. Is it only for Windows?

**Comment by bbondy** [comment]
I actually went back and forth about linux, does it not hide anything for all distros? I noticed the electron code only ignores OS X.

**Comment by luixxiul** [comment]
Turning on that option does not hide the bookmark toolbar. Rather than that I'm not sure what it is supposed to do because it is unclear to me where the menu bar is.

**Comment by bsclifton** [comment]
Interesting... is the menu in BrowserWindow a Windows only feature? 

It's supposed to hide this menu (see picture) which is toggleable by hitting alt (at least on Windows) and hidden by default (before this PR- which sets it to shown for Windows by default):
menu

**Comment by luixxiul** [comment]
@bsclifton On Unity (Ubuntu) like this:



Not sure about other desktop environments.

**Comment by bsclifton** [comment]
ah- it seems at least with Ubuntu/Unity, the menu ALWAYS shows

**Comment by bbondy** [comment]
I'll post separate t fix for Ubuntu/Unity but probably not the highest priority.  The option is respected and works properly in Dedbian with Gnome.

**Pull request closed without merging**